internal/p2p/pqueue.go

```diff
@@ -290,3 +274,14 @@ func (s *pqScheduler) push(pqEnv *pqEnvelope) {
 		s.sizes[uint(s.chDescs[i].Priority)] += pqEnv.size
 	}
 }
+func (s *pqScheduler) close() {
```

Comment:
Is the point here to handle double-closes gracefully? If so—a possible alternative to consider:

   signalCh <-chan struct{}
   close    func()

   sigCtx, cancel := context.WithCancel(ctx)
   // ... 
   signalCh: sigCtx.Done(),
   close:    cancel,
   
which would let `closed` work the same way and `close` would not need its own method, and multiple cancels are already handled. YMMV, though.

```suggestion

func (s *pqScheduler) close() {
```